compiler/crates/relay-typegen/src/visit.rs

```diff
@@ -1618,12 +1666,21 @@ fn make_prop(
                         type_,
                     )
                 });
-
+                let deprecated = linked_field
+                    .deprecation
+                    .as_ref()
+                    .map(|s| s.reason.as_ref().map(|r| r.to_string()));
+                let mut deprecated_reason: Option<String> = None;
+                if deprecated.is_some() && deprecated.as_ref().unwrap().is_some() {
+                    deprecated_reason = deprecated.unwrap();
+                }
```

Comment:
In general you should not need to do any unwrapping here. I think some kind of map/and_then chain would work (may need to fiddle with ownership a bit)

Also you might find this interesting: https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html

```suggestion
                let deprecated_reason = linked_field
                    .deprecation
                    .and_then(|s| s.reason)
                    .map(|reason| reason.to_string());
```